pkg/registry/core/pod/strategy.go

```diff
@@ -97,6 +99,31 @@ func (podStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object
 	oldPod := old.(*api.Pod)
 	newPod.Status = oldPod.Status
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		for i, c := range newPod.Spec.Containers {
+			if c.Resources.Requests == nil {
+				continue
+			}
+			if diff.ObjectDiff(oldPod.Spec.Containers[i].Resources, c.Resources) == "" {
+				continue
+			}
+			findContainerStatus := func(css []api.ContainerStatus, cName string) (api.ContainerStatus, bool) {
+				for i := range css {
+					if css[i].Name == cName {
+						return css[i], true
+					}
+				}
+				return api.ContainerStatus{}, false
+			}
+			if cs, ok := findContainerStatus(newPod.Status.ContainerStatuses, c.Name); ok {
+				if diff.ObjectDiff(c.Resources.Requests, cs.ResourcesAllocated) != "" {
+					newPod.Status.Resize = api.PodResizeStatusProposed
+					break
```

Comment:
Is this the right place for this logic?

This function says its purpose is `// PrepareForUpdate clears fields that are not allowed to be set by end users on update.` but this is actually performing the update which could break guarantees/assumptions from callers.